_posts/2019-01-01-install-kubernetes-kubeadm.md

```diff
@@ -0,0 +1,500 @@
+```
+kubeadm init --apiserver-advertise-address=<server_ip> --pod-network-cidr=10.244.0.0/16
+```
+Replace the **server_ip** above with the IP of your master. The other supporting cluster services/nodes will connect to this IP while forming the cluster. For example in our case the it looks like 
+```
+[root@kubernetes1 ~]# kubeadm init --apiserver-advertise-address=192.168.37.48 --pod-network-cidr=10.244.0.0/16
+```
+
+Once the command is successful, you will get a token which other nodes will use to join to the cluster. This will be displayed on your terminal and will look something like
+```
+kubeadm join 192.168.37.48:6443 --token 8hp10q.i4ln2b1ogof374aj --discovery-token-ca-cert-hash sha256:6a59c9b03fa971aef94d61a4f4c1a6b085308f88aab4db1a4affda8d65987867
+```
+You should carefully save this token. Your Kubernetes master should be initialized successfully by now. 
+
+### Create kube configuration in home directory
+To start using your cluster, you need to run the following as a regular user.
+```
+mkdir -p $HOME/.kube
+sudo cp -i /etc/kubernetes/admin.conf $HOME/.kube/config
+sudo chown $(id -u):$(id -g) $HOME/.kube/config
+```
+
+### Install flannel network
+You must deploy a pod network before anything will actually function properly. We will next, deploy the flannel network to the kubernetes cluster using the kubectl command. To check other networking options that can be used with kubernetes, please have a look [here](https://kubernetes.io/docs/concepts/cluster-administration/networking/).
+```
+kubectl apply -f https://raw.githubusercontent.com/coreos/flannel/master/Documentation/kube-flannel.yml
+```
+Check if all the components are deployed properly
+```
+[root@kubernetes1 ~]# kubectl get nodes
+NAME                        STATUS     ROLES    AGE     VERSION
+kubernetes1.oslo.sysco.no   Ready      master   7m18s   v1.13.1
+```
+Also check if all required pods are running properly.
+```
+[root@kubernetes1 ~]# kubectl get pods --all-namespaces
+NAMESPACE     NAME                                                READY   STATUS    RESTARTS   AGE
+kube-system   coredns-86c58d9df4-kjmxn                            1/1     Running   0          63m
+kube-system   coredns-86c58d9df4-lfx4p                            1/1     Running   0          63m
+kube-system   etcd-kubernetes1.oslo.sysco.no                      1/1     Running   0          62m
+kube-system   kube-apiserver-kubernetes1.oslo.sysco.no            1/1     Running   0          62m
+kube-system   kube-controller-manager-kubernetes1.oslo.sysco.no   1/1     Running   0          63m
+kube-system   kube-flannel-ds-amd64-mjn4j                         1/1     Running   0          55m
+kube-system   kube-proxy-w8rfz                                    1/1     Running   0          55m
+kube-system   kube-scheduler-kubernetes1.oslo.sysco.no            1/1     Running   0          62m
+```
+
+### Disable existing firewall service
```

Comment:
Why do we have to disable existing firewall (https://github.com/sysco-middleware/sysco-middleware.github.io/blob/install-kuberneter-kubeadm/_posts/2019-01-01-install-kubernetes-kubeadm.md#disable-existing-firewall-service)? It sounds like a bad idea, but I assume there are good reasons for that (and that the system security is still taken care of). Either one sentence explaining why or a link would be good to have here.

Reply:
Hi, 

You are correct. Disabling of the firewall was a bad idea. I did some research and found the correct way.

I have renamed the section to "Configure correct firewall rules", instead of just disabling the firewall rules. Please let me know if this looks good.